elasticsearch/bulk/bulk.go

```diff
@@ -190,25 +216,49 @@ func (b *Bulk) flushMessages() {
 		}
 		b.batchTicker.Reset(b.batchTickerDuration)
 		b.batch = b.batch[:0]
+		b.batchKeys = make(map[string]int, b.batchSizeLimit)
+		b.batchIndex = 0
```

Comment:
0 is meaningful in terms of index. Is it really necessary to assign this value to 0? If not we can assign it with like -1 :)

Reply:
It's necessary because my goal is to get it back on index 0.